docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,448 @@
+
+Instead, we can set up the repository determine the configuration for resources based on the environment:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v4.py startafter=start endbefore=end
+# repository.py
+
+# Note: there are multiple issues with how this config is specified, mainly that
+# passwords are being stored in code. This will be addressed next.
+@repository
+def repo():
+    resource_defs = {
+        "local": {
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "me@company.com",
+                    # password in config is bad practice
+                    "password": "my_super_secret_password",
+                    "database": "LOCAL",
+                    "schema": "ALICE",
+                }
+            ),
+        },
+        "production": {
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "dev@company.com",
+                    # password in config is bad practice
+                    "password": "company_super_secret_password",
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS",
+                }
+            ),
+        },
+    }
+    deployment_name = os.getenv("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources(
+            [items, comments, stories], resource_defs=resource_defs[deployment_name]
+        )
+    ]
+```
+
+Now, we can set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<Note>
+If you're using <a href="https://dagster.io/cloud">Dagster Cloud</a>, this environment variable will be set for you. Additionally, if you're using <a href="https://docs.dagster.cloud/guides/branch-deployments">Branch Deployments</a> <code>DAGSTER_DEPLOYMENT</code> will be automatically set to <code>VAR</code>.
+
+</Note>
+
+We still have some problems with this setup:
+
+1. Developers needs to remember to change `user` and `password` to their credentials and `schema` to their name when developing locally.
+2. Passwords are being stored in code.
+
+We can easily solve these problems because the Snowflake IO manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables, and point Dagster to those environment variables in the run configuration:
```

Comment:
```suggestion
We can easily solve these problems because the Snowflake I/O manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables and point Dagster to those environment variables in the run configuration:
```

- `IO` > `I/O`
- Remove unnecessary comma